### ISSUE TITLE
Support encrypted ssh keys

### ISSUE THREAD

**Issue by toby** [charmbracelet/charm#107]
Currently only unencrypted SSH keys are supported, we should also provide support for password encrypted keys.

**Comment by aymanbagabas** [comment]
FWIW, [keygen](https://github.com/charmbracelet/keygen) now supports password-protected keys.

**Comment by bashbunni** [comment]
Also sk-protected keys at some point :eyes:

**Comment by bashbunni** [comment]
Hey @aymanbagabas can I close this issue?

**Comment by aymanbagabas** [comment]
> Hey @aymanbagabas can I close this issue?

No, not yet. The mentioned pr only updates the dependency which now supports encrypted keys but charm still does not support encrypted keys yet.

**Comment by caarlos0** [comment]
Hey, the charm cloud is discontinued, so this is out of our roadmap.

https://github.com/charmbracelet/charm?tab=readme-ov-file#sunsetting-charm-cloud

Closing the issue, but feel free to comment if you have further questions.
